src/components/TimePicker/TimePicker.js

```diff
@@ -0,0 +1,487 @@
+}
+
+// if we need manually to move selection to the left we need to decrease both selection start and end by one
+function decreaseBothSelectionByOne({start, end}) {
+    if (start === 0) {
+        return {start: 0, end: 0};
+    }
+    return {start: start - 1, end: end - 1};
+}
+
+function replaceWithZeroAtPosition(originalString, position) {
+    if (position === 0 || position > 2) {
+        return originalString;
+    }
+    return `${originalString.slice(0, position - 1)}0${originalString.slice(position)}`;
+}
+
+function TimePicker({forwardedRef, defaultValue, onSubmit, onInputChange}) {
+    const {numberFormat, translate} = useLocalize();
+    const {isExtraSmallScreenHeight} = useWindowDimensions();
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const value = DateUtils.extractTime12Hour(defaultValue);
+    const canUseTouchScreen = DeviceCapabilities.canUseTouchScreen();
+
+    const [isError, setError] = useState(false);
+    const [selectionHour, setSelectionHour] = useState({start: 0, end: 0});
+    const [selectionMinute, setSelectionMinute] = useState({start: 2, end: 2}); // we focus it by default so need  to have selection on the end
+    const [hours, setHours] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).hour);
+    const [minute, setMinute] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).minute);
+    const [amPmValue, setAmPmValue] = useState(() => DateUtils.get12HourTimeObjectFromDate(value).period);
+
+    const hourInputRef = useRef(null);
+    const minuteInputRef = useRef(null);
+
+    const focusMinuteInputOnFirstCharacter = useCallback(() => {
+        const cleanupTimer = setSelection({start: 0, end: 0}, minuteInputRef, setSelectionMinute);
+        return cleanupTimer;
+    }, []);
+    const focusHourInputOnLastCharacter = useCallback(() => {
+        setSelectionHour({start: 2, end: 2});
+        const timer = setTimeout(() => {
+            hourInputRef.current.focus();
+        }, 10);
+        return () => clearTimeout(timer);
+    }, []);
+
+    const validate = useCallback(
+        (time) => {
+            const isValid = DateUtils.isTimeAtLeastOneMinuteInFuture({timeString: time || `${hours}:${minute} ${amPmValue}`, dateTimeString: defaultValue});
+            setError(!isValid);
+            return isValid;
+        },
+        [hours, minute, amPmValue, defaultValue],
+    );
+
+    // This function receive value from hour input and validate it
+    // The valid format is HH(from 00 to 12). If the user input 9, it will be 09. If user try to change 09 to 19 it would skip the first character
+    const handleHourChange = (text) => {
+        const isOnlyNumericValue = /^\d+$/.test(text.trim());
+        // Skip if the user is pasting the text or use non numeric characters.
+        if (selectionHour.start !== selectionHour.end || !isOnlyNumericValue) {
+            return;
+        }
```

Comment:
This made the user unable to select all and delete (selection range have a start that is different than the end) - Coming from https://github.com/Expensify/App/issues/33143

Reply:
It was made intentionally as it brings a new level of complexity, and validating becomes messy. We decided to release it as is and change how this input would work in further PRs